### PR TITLE
Chronos: add best validation mode to forecasters' fit

### PR DESCRIPTION
**Pull request by Chaselzxy**
## Description

1. Why the change?
To add best validation mode to forecasters' fit. Forecasters will monitor the validation_loss and And load the checkpoint of the epoch with the smallest val_loss after the training in this mode.
https://github.com/intel-analytics/BigDL/issues/4958

2. User API changes
Add a mode best validation for validation_mode.

3. Summary of the change
Change the file [python/chronos/src/bigdl/chronos/forecaster/base_forecaster.py](https://github.com/intel-analytics/BigDL/pull/5290/files#diff-f90b2e349024be33300085266c59def4560dd7b81de6573ad6b93a51fd382fb0).

4. How to test?
Add unit test in
[python/chronos/test/bigdl/chronos/forecaster/test_lstm_forecaster.py](https://github.com/intel-analytics/BigDL/pull/5290/files#diff-802bb8b90c18a761e7e0d4eb000cebddfde5366a48cb7201533a4ca4d0c295ce)
[python/chronos/test/bigdl/chronos/forecaster/test_nbeats_forecaster.py](https://github.com/intel-analytics/BigDL/pull/5290/files#diff-2d759068533e4a21b1b0e56c77e144731d37b63ff3ee83f7dd0721dce5c62458)
[python/chronos/test/bigdl/chronos/forecaster/test_seq2seq_forecaster.py](https://github.com/intel-analytics/BigDL/pull/5290/files#diff-7f8329eefc66b1ba67d3c33380fff7e3d7134181812f20b197f53964616e3337)
[python/chronos/test/bigdl/chronos/forecaster/test_tcn_forecaster.py](https://github.com/intel-analytics/BigDL/pull/5290/files#diff-779e0b1ffaba1d91d35970801d1c9c81958393d710ea1d6286913c421d76c2c7)

5. New dependencies

### CODE REVIEW

File: python/chronos/src/bigdl/chronos/forecaster/base_forecaster.py [hunk 1]
```diff
@@ -263,6 +263,10 @@ def fit(self, data, validation_data=None, epochs=1, batch_size=32, validation_mo
                |
                | 2. earlystop:
                | Monitor the val_loss and stop training when it stops improving.
+               |
+               | 3. best validation:
```
Comment (TheaperDeng) [inline python/chronos/src/bigdl/chronos/forecaster/base_forecaster.py:267]: best_epoch

File: python/chronos/src/bigdl/chronos/forecaster/base_forecaster.py [hunk 2]
```diff
@@ -334,7 +338,14 @@ def fit(self, data, validation_data=None, epochs=1, batch_size=32, validation_mo
                                                                      name="forecaster_tmp_log")
             from pytorch_lightning.callbacks import EarlyStopping
             early_stopping = EarlyStopping('val/loss', patience=earlystop_patience)
-            callbacks = [early_stopping] if validation_mode == 'earlystop' else None
+            from pytorch_lightning.callbacks import ModelCheckpoint
+            checkpoint_callback = ModelCheckpoint(monitor="val/loss", dirpath='validation',
+                                                  filename='best', save_on_train_epoch_end=True)
```
Comment (TheaperDeng) [inline python/chronos/src/bigdl/chronos/forecaster/base_forecaster.py:343]: nothing to comment here, that's really a good API for this function

**Review by TheaperDeng — COMMENTED** [review]
others LGTM

**Review by TheaperDeng — APPROVED** [review]
LGTM, pls run jenkins

**Comment by Chaselzxy** [comment]
http://10.112.231.51:18889/view/BigDL-PR-Validation/job/BigDL-Chronos-PR-Validation/660/

**Pull request merged**